src/base/ParentWorker.ts

```diff
@@ -68,47 +70,103 @@ export default class ParentWorker extends Worker<BaseJobParams, BaseJobResult> {
     if (jobs.length === 0) {
       const transaction = this.nonBlockingRedis.multi();
       const newJobs: string[] = [];
-      params.forEach((param) => {
-        if (!param.childName) {
-          this.logger.warn(
-            "Child name is missing in child params",
-            propertiesToLog
+      await Promise.all(
+        params.map(async (param) => {
+          if (!param.childName) {
+            this.logger.warn(
+              "Child name is missing in child params",
+              propertiesToLog
+            );
+            return;
+          }
+
+          const priority = param.priority || job.priority;
+
+          // the child flow name is composed of the parent queue name and the child name
+          const childFlowName = `${parentQueueName}:${param.childName}`;
+
+          // generate child job id
+          const childJobId = generateJobId(childFlowName);
+
+          const childJobKey = keyFormatter.job(childJobId);
+
+          const childQueueKey = keyFormatter.childWaitingQueueName(
+            parentQueueName,
+            param.childName,
+            priority
           );
-          return;
-        }
 
-        // the child flow name is composed of the parent queue name and the child name
-        const childFlowName = `${parentQueueName}:${param.childName}`;
-
-        // generate child job id
-        const childJobId = generateJobId(childFlowName);
-
-        const childJobKey = keyFormatter.job(childJobId);
-
-        const childQueueKey = keyFormatter.childWaitingQueueName(
-          parentQueueName,
-          param.childName,
-          param.priority || job.priority
-        );
+          const childJob = param;
+          childJob.priority = priority;
+          childJob.flowName = childFlowName;
+          childJob.correlationId = job.correlationId;
+
+          const isRiverJob = param.childName === "discord";
+
+          // insert riverjob to postgres
+          if (isRiverJob) {
+            const result = await this.queueClient.postgresClient.query(
+              `INSERT INTO river_job (
+                      state,
+                      attempt,
+                      max_attempts,
```

Comment:
We could use these as a static, but maybe later on.

Reply:
Absolutely, but we are really in a transition phase rn